### PR TITLE
For #7673: Move DownloadProgress and Status to DownloadState

### PR DESCRIPTION
**Pull request by kglazko**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt [hunk 1]
```diff
@@ -28,17 +30,27 @@ import kotlin.random.Random
...
         Environment.getExternalStoragePublicDirectory(destinationDirectory).path + "/" + fileName
 }
+
+enum class DownloadJobStatus {
+    ACTIVE,
+    PAUSED,
+    CANCELLED,
+    FAILED,
+    COMPLETED
+}
```
Comment (Amejia481) [inline components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:56]: We can move it inside of  `DownloadState` and rename it to `Status`. This way when we want to access will it be `DownloadState.Status`. Let's also add docs indicating its purpose and each status :)

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 2]
```diff
@@ -133,27 +127,16 @@ abstract class AbstractFetchDownloadService : Service() {
     internal fun setDownloadJobStatus(downloadJobState: DownloadJobState, status: DownloadJobStatus) {
         synchronized(context) {
             if (status == DownloadJobStatus.ACTIVE) { downloadJobState.notifiedStopped = false }
-            downloadJobState.status = status
+            downloadJobState.state.status = status
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:130]: You also have to update the store, you could use  [updateDownloadState](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L677) this [here you can see an example](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L612)

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt [hunk 3]
```diff
@@ -28,17 +30,27 @@ import kotlin.random.Random
...
-    val fileName: String? = null,
-    val contentType: String? = null,
-    val contentLength: Long? = null,
-    val userAgent: String? = null,
-    val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
-    val referrerUrl: String? = null,
-    val skipConfirmation: Boolean = false,
-    val id: Long = Random.nextLong(),
-    val sessionId: String? = null
+        val url: String,
```
Comment (Amejia481) [inline components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:33]: Let's not add the extra indentation

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt [hunk 4]
```diff
@@ -28,17 +30,27 @@ import kotlin.random.Random
...
+        val fileName: String? = null,
+        val contentType: String? = null,
+        val contentLength: Long? = null,
+        var currentBytesCopied: Long = 0,
```
Comment (Amejia481) [inline components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:37]: Let's add docs :)

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt [hunk 5]
```diff
@@ -28,17 +30,27 @@ import kotlin.random.Random
...
+        @GuardedBy("context") var status: DownloadJobStatus = DownloadJobStatus.ACTIVE,
```
Comment (Amejia481) [inline components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:38]: Let's add docs :)
I think we can remove the `@GuardedBy` annotation

Comment (Amejia481) [reply components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt]: We want this values to be `val` as we want them to be immutable.

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt (hunk 4 above)
Comment (Amejia481) [reply components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:37]: We want this values to be `val` as we want it to be immutable.

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 6]
```diff
@@ -596,43 +585,43 @@ abstract class AbstractFetchDownloadService : Service() {
...
 
         // To ensure that we copy all files (even ones that don't have fileSize, we must NOT check < fileSize
-        while (getDownloadJobStatus(downloadJobState) == DownloadJobStatus.ACTIVE) {
+        while (getDownloadJobStatus(downloadJobState) == Status.ACTIVE) {
             val bytesRead = inStream.read(data)
 
             // If bytesRead is -1, there's no data left to read from the stream
             if (bytesRead == -1) { break }
 
-            downloadJobState.currentBytesCopied += bytesRead
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:630]: You also have to update the store here, as we want this changes to be reflected there too, you could use  [updateDownloadState](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L677) this [here you can see an example](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L612)

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt (hunk 5 above)
Comment (kglazko) [reply components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt]: Do you think it's ok to set it as Active right away? It wouldn't let me not set it to anything like before.

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 7]
```diff
@@ -596,43 +585,43 @@ abstract class AbstractFetchDownloadService : Service() {
...
+            downloadJobState.state.currentBytesCopied += bytesRead
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:619]: You also have to update the store here, as we want this changes to be reflected there too, you could use  [updateDownloadState](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L677) this [here you can see an example](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L612)

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt (hunk 2 above)
Comment (kglazko) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:130]: Added the update store but GH isn't picking up the changes

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt (hunk 7 above)
Comment (kglazko) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:619]: Added the update store but GH isn't picking up the changes

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt [hunk 8]
```diff
@@ -28,17 +30,28 @@ import kotlin.random.Random
...
-    val referrerUrl: String? = null,
-    val skipConfirmation: Boolean = false,
-    val id: Long = Random.nextLong(),
-    val sessionId: String? = null
+        val url: String,
+        val fileName: String? = null,
+        val contentType: String? = null,
+        val contentLength: Long? = null,
+        val currentBytesCopied: Long = 0,
+        @GuardedBy("context") var status: Status = Status.ACTIVE,
```
Comment (Amejia481) [inline components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:38]: Let's remove the annotation and use val

File: components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt [hunk 9]
```diff
@@ -28,17 +30,28 @@ import kotlin.random.Random
...
+        val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
+        val referrerUrl: String? = null,
+        val skipConfirmation: Boolean = false,
+        val id: Long = Random.nextLong(),
+        val sessionId: String? = null
 ) : Parcelable {
     val filePath: String get() =
         Environment.getExternalStoragePublicDirectory(destinationDirectory).path + "/" + fileName
+
+    enum class Status {
```
Comment (Amejia481) [inline components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt:49]: Nit: Let's add docs describing each status

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 10]
```diff
@@ -248,13 +248,10 @@ abstract class AbstractFetchDownloadService : Service() {
 
         // Create a new job and add it, with its downloadState to the map
         val downloadJobState = DownloadJobState(
-            state = download,
+            state = download.copy(status = DOWNLOADING),
             foregroundServiceId = foregroundServiceId
         )
 
-        val newState = downloadJobState.state.copy(status = Status.DOWNLOADING)
-        updateDownloadState(newState)
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt]: @kglazko this is the change that I commented you about on slack. We were trying to use `updateDownloadState` to early as `updateDownloadState` update the state on `downloadJobs` but downloadJobState gets added much after in line 259

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/AbstractFetchDownloadServiceTest.kt [hunk 11]
```diff
@@ -957,6 +958,10 @@ class AbstractFetchDownloadServiceTest {
         browserStore.dispatch(DownloadAction.QueueDownloadAction(download)).joinBlocking()
         service.registerNotificationActionsReceiver()
         service.onStartCommand(downloadIntent, 0, 0)
+        service.downloadJobs.values.forEach {
+            println("TESTING: " + it.job!!.isActive .toString())
+        }
```
Comment (Amejia481) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/AbstractFetchDownloadServiceTest.kt:963]: nit: I think we don't need this line :)

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadDialogFragmentTest.kt [hunk 12]
```diff
@@ -22,10 +22,12 @@ class DownloadDialogFragmentTest {
     @Before
     fun setup() {
         dialog = object : DownloadDialogFragment() {}
+        val currentBytes = ("Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML" +
+                ", like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36").length.toLong()
         download = DownloadState(
             "http://ipv4.download.thinkbroadband.com/5MB.zip",
             "5MB.zip", "application/zip", 5242880,
-            "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"
+                currentBytes
```
Comment (Amejia481) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadDialogFragmentTest.kt:30]: Nit: I think this value was for the user agent, the `currentBytesCopied` is optional, we are not required to pass a value. 
`                userAgent = "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"`

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadNotificationTest.kt [hunk 13]
```diff
@@ -29,9 +23,9 @@ class DownloadNotificationTest {
     fun getProgress() {
         val downloadJobState = DownloadJobState(
                 job = null,
-                state = DownloadState(url = "mozilla.org/mozilla.txt", contentLength = 100L),
-                currentBytesCopied = 10,
-                status = ACTIVE,
+                state = DownloadState(url = "mozilla.org/mozilla.txt", contentLength = 100L,
+                        currentBytesCopied = 10,
+                        status = DownloadState.Status.DOWNLOADING),
```
Comment (Amejia481) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadNotificationTest.kt:28]: nit: We can use import `DownloadState.Status.DOWNLOADING` and just use `DOWNLOADING`

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/SimpleDownloadDialogFragmentTest.kt [hunk 14]
```diff
@@ -37,10 +37,12 @@ class SimpleDownloadDialogFragmentTest {
     @Before
     fun setup() {
         mockFragmentManager = mock()
+        val currentBytesCopied = ("Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko)" +
+                " Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36").length.toLong()
         download = DownloadState(
             "http://ipv4.download.thinkbroadband.com/5MB.zip",
             "5MB.zip", "application/zip", 5242880,
-            "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"
+            currentBytesCopied = currentBytesCopied
```
Comment (Amejia481) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/SimpleDownloadDialogFragmentTest.kt:45]: nit: This value was for the `userAgent`, `currentBytesCopied` is an optional value, we don't have to provide it, if we don't need it

`userAgent = "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"`

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/manager/AndroidDownloadManagerTest.kt [hunk 15]
```diff
@@ -37,10 +36,12 @@ class AndroidDownloadManagerTest {
 
     @Before
     fun setup() {
+        val currentBytesCopied = ("Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko)" +
+                " Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36").length.toLong()
         download = DownloadState(
             "http://ipv4.download.thinkbroadband.com/5MB.zip",
             "", "application/zip", 5242880,
-            "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"
+            currentBytesCopied
```
Comment (Amejia481) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/manager/AndroidDownloadManagerTest.kt:44]: Nit: the same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459114027

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/manager/FetchDownloadManagerTest.kt [hunk 16]
```diff
@@ -51,7 +50,8 @@ class FetchDownloadManagerTest {
         download = DownloadState(
             "http://ipv4.download.thinkbroadband.com/5MB.zip",
             "", "application/zip", 5242880,
-            "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"
+            ("Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko)" +
+                    " Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36").length.toLong()
```
Comment (Amejia481) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/manager/FetchDownloadManagerTest.kt:54]: Nit: the same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459114027

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 17]
```diff
@@ -220,12 +214,13 @@ abstract class AbstractFetchDownloadService : Service() {
...
                                 contentType = currentDownloadJobState.state.contentType
-                            )) {
+                            )
+                        ) {
                             val fileExt = MimeTypeMap.getFileExtensionFromUrl(
-                                    currentDownloadJobState.state.filePath.toString()
+                                currentDownloadJobState.state.filePath.toString()
                             )
                             val errorMessage = applicationContext.getString(
-                                    R.string.mozac_feature_downloads_open_not_supported1, fileExt
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:228]: Please don't commit changes in indentation in lines that you don't intentionally modified as it makes the diff way harder to read. :)

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 18]
```diff
@@ -390,22 +387,22 @@ abstract class AbstractFetchDownloadService : Service() {
...
-                    title = fileName,
-                    description = fileName,
-                    isMediaScannerScannable = true,
-                    mimeType = download.contentType ?: "*/*",
-                    path = file.absolutePath,
-                    length = download.contentLength ?: file.length(),
-                    // Only show notifications if our channel is blocked
-                    showNotification = !DownloadNotification.isChannelEnabled(context),
-                    uri = url,
-                    referer = download.referrerUrl?.toUri()
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:408]: Same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459538893

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 19]
```diff
@@ -454,7 +451,8 @@ abstract class AbstractFetchDownloadService : Service() {
     internal fun updateNotificationGroup(): Notification? {
         return if (SDK_INT >= Build.VERSION_CODES.N) {
             val downloadList = downloadJobs.values.toList()
-            val notificationGroup = DownloadNotification.createDownloadGroupNotification(context, downloadList)
+            val notificationGroup =
+                DownloadNotification.createDownloadGroupNotification(context, downloadList)
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:455]: Same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459538893

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 20]
```diff
@@ -500,7 +499,9 @@ abstract class AbstractFetchDownloadService : Service() {
             previousDownload = downloadJobs.values.firstOrNull {
                 it.foregroundServiceId == compatForegroundNotificationId
             }
-            downloadJobState.foregroundServiceId to createCompactForegroundNotification(downloadJobState)
+            downloadJobState.foregroundServiceId to createCompactForegroundNotification(
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:502]: Same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459538893

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 21]
```diff
@@ -596,43 +597,50 @@ abstract class AbstractFetchDownloadService : Service() {
...
+                downloadJobState.state.currentBytesCopied + bytesRead
+            )
+            updateDownloadState(newState)
 
             outStream.write(data, 0, bytesRead)
         }
-        logger.debug("Finishing copyInChunks ${downloadJobState.state.url} " +
-                "currentBytesCopied ${downloadJobState.currentBytesCopied}")
+        logger.debug(
+            "Finishing copyInChunks ${downloadJobState.state.url} " +
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:641]: Same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459538893

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 22]
```diff
@@ -687,13 +695,17 @@ abstract class AbstractFetchDownloadService : Service() {
         download: DownloadState,
         append: Boolean
     ): DownloadState {
-        if (append) { return download }
+        if (append) {
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:698]: Same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459538893

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 23]
```diff
@@ -687,13 +695,17 @@ abstract class AbstractFetchDownloadService : Service() {
...
+            return download
+        }
 
         return download.fileName?.let {
-            download.copy(fileName = DownloadUtils.uniqueFileName(
-                Environment.getExternalStoragePublicDirectory(download.destinationDirectory),
-                it
-            ))
+            download.copy(
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:703]: Same as https://github.com/mozilla-mobile/android-components/pull/7698/files#r459538893

**Comment by kglazko** [comment]
After testing with Fenix, this requires some Fenix-side changes which I've addressed in this PR: https://github.com/mozilla-mobile/fenix/pull/12921

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt (hunk 17 above)
Comment (kglazko) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:228]: Sorry, my Android Studio somehow became incorrectly configured with indentations/spacing after the last update and I'm not sure why that happened or how I can reset it to what we need :(

Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:228]: Don't worry, it looks like it was applying the styling to the whole file, it's better to just apply styling to the sections that you change, one way to verify it before pushing just review the changes and make sure that only the parts that you changed are getting committed, that make the review process easier :)

File: docs/changelog.md [hunk 24]
```diff
@@ -23,6 +23,9 @@ permalink: /changelog/
 
 * **feature-syncedtabs**
   * ⚠️ **This is a breaking change**: Adds context to the constructor of `SyncedTabsFeature`.
+  * ⚠️ **This is a breaking change**: Removed the following properties from `DownloadJobState` in `AbstractFetchDownloadService` and added to `DownloadState`: `DownloadJobStatus` (now renamed to `DownloadStatus`) and `currentBytesCopied`. These properties can now be read from `DownloadState`.
+    * ⚠️ **This is a breaking change**: Removed the enum class `DownloadJobStatus` from `AbstractFetchDownloadService` and moved into `DownloadState`, and removed the `ACTIVE` state while introducing two new states called `INITIATED` and `DOWNLOADING`.
```
Comment (Amejia481) [inline docs/changelog.md:27]: The description looks  good 👍  lets add another section for * **feature-downloads** as this section belongs to **feature-syncedtabs**

Comment (kglazko) [reply docs/changelog.md:29]: Ok, is it good now?

Comment (Amejia481) [reply docs/changelog.md:29]: Looking good, let's just remove the extra indentation on line 27 `    * ⚠️ **This is a breaking change**:` :)
image

**Comment by kglazko** [comment]
bors r=arturo

**Pull request merged**